impl/maven-core/src/main/java/org/apache/maven/project/MavenProject.java

```diff
@@ -861,7 +861,7 @@ public void addLicense(License license) {
     }
 
     public void setArtifacts(Set<Artifact> artifacts) {
-        this.artifacts = artifacts;
```

Comment:
current tests fail to make field `final`.
this need be be investigated further.

Reply:
wrong impl. of course. should be in impl. detail if final or not.